core/src/main/java/com/netflix/conductor/core/events/EventProcessor.java

```diff
@@ -1,235 +1,247 @@
+                    }
+                }
+
+                CompletableFuture<List<Pair<EventExecution, Boolean>>> allFutures = CompletableFutures.allAsList(futuresList);
+                allFutures.whenComplete((result, error) -> result.forEach(eventExecutionBooleanPair -> {
+                    if (error != null || !eventExecutionBooleanPair.getValue()) {
+                        executionService.removeEventExecution(eventExecutionBooleanPair.getKey());
+                        failedExecutions.add(eventExecutionBooleanPair.getKey());
+                    } else {
+                        executionService.updateEventExecution(eventExecutionBooleanPair.getKey());
+                    }
+                })).get();
+            }
+
+            if (failedExecutions.isEmpty()) {
+                queue.ack(Collections.singletonList(msg));
+            } else if (queue.rePublishIfNoAck()) {
+                // re-submit this message to the queue, to be retried later
+                // This is needed for queues with no unack timeout, since messages are removed from the queue
+                queue.publish(Collections.singletonList(msg));
+            }
+
+        } catch (Exception e) {
+            logger.error("Error handling message: {} on queue:{}", msg, queue.getName(), e);
+        }
+    }
+
+    @SuppressWarnings("Guava")
+    @VisibleForTesting
+    Pair<EventExecution, Boolean> execute(EventExecution eventExecution, Action action, String payload) {
+        try {
+            String methodName = "executeEventAction";
+            String description = String.format("Executing action: %s for event: %s with messageId: %s with payload: %s", action.getAction(), eventExecution.getId(), eventExecution.getMessageId(), payload);
+            logger.debug(description);
+
+            Predicate<Throwable> filterException = throwableException -> {
+                if (throwableException != null) {
+                    return !(throwableException.getCause() instanceof UnsupportedOperationException);
+                }
+                return true;
+            };
+            Map<String, Object> output = new RetryUtil<Map<String, Object>>().retryOnException(() -> actionProcessor.execute(action, payload, eventExecution.getEvent(), eventExecution.getMessageId()),
+                    filterException, null, RETRY_COUNT, description, methodName);
+            if (output != null) {
+                eventExecution.getOutput().putAll(output);
+            }
+            eventExecution.setStatus(Status.COMPLETED);
+        } catch (RuntimeException e) {
+            logger.error("Error executing action: {} for event: {} with messageId: {}", action.getAction(), eventExecution.getEvent(), eventExecution.getMessageId(), e);
+            if (e.getCause() instanceof UnsupportedOperationException || (e.getCause() instanceof ApplicationException && ((ApplicationException) e.getCause()).getCode() != ApplicationException.Code.BACKEND_ERROR)) {
```

Comment:
filterException predicate should check transient ApplicationException as well, once it is moved to a private method, use it here as well.

Reply:
`INTERNAL_ERROR` is thrown in the server for non-transient exceptions like json parsing errors etc.